### ISSUE TITLE
bug: Master-slave environment，the slave error when UPDATE BIT type data in the master.

### ISSUE THREAD

**Issue by davidshiz** [stoneatom/stonedb#1190]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/stoneatom/stonedb/blob/stonedb-5.7-dev/CONTRIBUTING.md).

### Please confirm if bug report does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the problem

Execute the following sql in Master:
```sql
create table t2(a bit(2), b bit(2));
insert into t2 (a) values (0x01), (0x03), (0x02);
update t2 set b= a-1;
```

the slave error when UPDATE BIT type in the master
```shell
 Last_SQL_Error: Could not execute Update_rows event on table bit_test.t2; You have an error in your SQL syntax; 
check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1, 
Error_code: 1064; Can't find record in 't2', Error_code: 1032; handler error HA_ERR_END_OF_FILE; the event's master log mybinlog.000001, end_log_pos 1768
```

### How To Reproduce

Execute the following sql in Master:

```sql
create table t2(a bit(2), b bit(2));
insert into t2 (a) values (0x01), (0x03), (0x02);
update t2 set b= a-1;
```

### Environment

```shell
root@ub01:~# /stonedb57/install/bin/mysqld --version
/stonedb57/install/bin/mysqld Ver 5.7.36-StoneDB for Linux on x86_64 (build-)
build information as follow:
Repository address: https://github.com/hustjieke/stonedb.git:feat_support_bit_type_issue919
Branch name: feat_support_bit_type_issue919
Last commit ID: https://github.com/stoneatom/stonedb/commit/38982a599b4f302f63f1612206d8f74d1606098f
Last commit time: Date: Tue Dec 27 08:44:36 2022 +0000
Build time: Date: Thu Dec 29 13:08:25 CST 2022
[root@localhost ~]# cat /etc/system-release
CentOS Linux release 7.9.2009 (Core)
```

### Are you interested in submitting a PR to solve the problem?

- [ ] Yes, I will!

**Comment by davidshiz** [comment]
A similar problem occurs when DELETE BIT data

**Comment by hustjieke** [comment]
The same reason with #1182, I'll fix it in #1182.
ping @davidshiz .
